src/couch/src/couch_httpd_auth.erl

```diff
@@ -99,27 +99,32 @@ special_test_authentication_handler(Req) ->
 basic_name_pw(Req) ->
     AuthorizationHeader = header_value(Req, "Authorization"),
     case AuthorizationHeader of
-        "Basic " ++ Base64Value ->
-            try
-                re:split(
-                    base64:decode(Base64Value),
-                    ":",
-                    [{return, list}, {parts, 2}]
-                )
-            of
-                ["_", "_"] ->
-                    % special name and pass to be logged out
-                    nil;
-                [User, Pass] ->
-                    {User, Pass};
-                _ ->
-                    nil
-            catch
-                error:function_clause ->
-                    throw({bad_request, "Authorization header has invalid base64 value"})
-            end;
+        % undefined is sent during startup process
+        undefined ->
```

Comment:
This `undefined` appear to me as a magic value. I think we can flip the order of clauses to make it more readable and be closer to functionality of original version (In original version we ignore any unexpected values).

```
basic_name_pw(Req) ->
    AuthorizationHeader = header_value(Req, "Authorization"),
    case AuthorizationHeader of
        Header when is_list(Header) ->
            [Basic, Base64Value] = string:split(AuthorizationHeader, " "),
            ...
        _ ->
            nil
   end.
```

Reply:
Your approach makes sense. See https://github.com/apache/couchdb/pull/3637/commits/dd0b666e743f6466910b278000a865bb087da451.